### DISCUSSION TITLE
/digging-deeper/helpers.html

### DISCUSSION THREAD

**Discussion by giscus** [goravel/goravel#255] [Announcements]
# /digging-deeper/helpers.html

[[toc]] 可用方法 路径 ----------- -------------- -------------- path.App() (#path-app) path.Base() (#path-base) path.Config() (#path-config) path.Database() (#path-database) path.Stor...

https://www.goravel.dev/zh/digging-deeper/helpers.html

**Comment by rowjak** [comment]
in 1.15.9,
i'm create a helper that can be used in all view,
```
package helpers

import (
	"strings"
	"time"
)

var CurrentPath string

var ViewHelpers = map[string]any{
	"today": func() string {
		return time.Now().Format("02-01-2006")
	},
	"currentPath": func() string {
		return CurrentPath
	},
	"isActive": func(menuPath string) string {
		if CurrentPath == menuPath || (menuPath != "/" && strings.HasPrefix(CurrentPath, menuPath+"/")) {
			return "active"
		}
		return ""
	},
}

```
an in view i'm call it using : {{ call .h.today }}


but in 1.16,
i got an error:
```
Error #01: template: header.tmpl:136:31: executing "layout/header.tmpl" at <call .h.today>: error calling call: call of nil
```

**Reply by hwbrzzl** [reply]
Thanks for your feedback, there may be an upgrade in http/template. Could you search the error via Google first? Let me know if you can't find any useful information.

**Reply by rowjak** [reply]
i'm sorry, its' my mistake.. i forgot to register helpers in service providers like in 1.15.9,
thank you 🙏
